modules/apps/app-builder/app-builder-test/src/testFunctional/tests/AppBuilder.testcase

```diff
@@ -1849,6 +1849,90 @@ definition {
 		Navigator.closeWindow();
 	}
 
+	@description = "Reuse my object's existing fields when creating a new form view"
+	@priority = "5"
+	test ReuseExistingFormViewFields {
+
+		AppBuilderAdmin.addCustomObject(
+			customObjectName = "my custom object",
+			isCreatingFormView = "true",
+			isCreatingViaPlusIcon = "true");
+
+		FormViewBuilder.addFieldByDoubleClick(fieldType = "Text");
+		FormViewBuilder.setFieldLabel(fieldLabel = "Full name");
+		FormViewBuilder.setFieldRequired();
+		FormViewBuilder.closeSidebar();
+		FormViewBuilder.validateFieldLabelRequired(fieldLabel = "Full name");
+
+		FormViewBuilder.addFieldByDoubleClick(fieldType = "Text");
+		FormViewBuilder.setFieldLabel(fieldLabel = "Nickname");
+		FormViewBuilder.closeSidebar();
+
+		FormViewBuilder.addFieldByDoubleClick(fieldType = "Numeric");
+		FormViewBuilder.setFieldLabel(fieldLabel = "Number of children");
+		FormViewBuilder.closeSidebar();
+
+		FormViewBuilder.addFieldByDoubleClick(fieldType = "Select from List");
+		FormViewBuilder.setFieldLabel(fieldLabel = "Occupation");
+		for (var number : list "1,2,3") {
+			FormViewBuilder.editOptionsRow(
+				optionName = "Select${number}",
+				optionValue = "Select ${number}",
+				rowNumber = "${number}");
+		}
+		FormViewBuilder.closeSidebar();
+
+		FormViewBuilder.addFieldByDoubleClick(fieldType = "Date");
+		FormViewBuilder.setFieldLabel(fieldLabel = "Date of birth");
+		FormViewBuilder.setFieldRequired();
+		FormViewBuilder.closeSidebar();
+		FormViewBuilder.validateFieldLabelRequired(fieldLabel = "Date of birth");
+
+		FormViewBuilder.addFieldByDoubleClick(fieldType = "Multiple Selection");
+		FormViewBuilder.setFieldLabel(fieldLabel = "Hobbies");
+		for (var number : list "1,2,3") {
+			FormViewBuilder.editOptionsRow(
+				optionName = "Multiple${number}",
+				optionValue = "Multiple ${number}",
+				rowNumber = "${number}");
+		}
+		FormViewBuilder.closeSidebar();
+
+		FormViewBuilder.setName(formViewName = "Registration Form");
+
+		FormViewBuilder.save();
+
+		AppBuilderAdmin.add();
+
+		FormViewBuilder.addFormViewAndValidateFieldLabel(
+			fieldType = "Text",
+			formViewName = "My Form View");
+
+		AppBuilderAdmin.openItem(itemName = "My Form View");
+
+		FormViewBuilder.addFromObjectField(fieldName = "Hobbies");
+
+		FormViewBuilder.setFieldLabel(fieldLabel = "Hobbies edited");
+
+		FormViewBuilder.validateFormMultipleSelectionField(
+			fieldLabel = "Multiple 1",
+			fieldName = "Hobbies");
+
+		FormViewBuilder.editFieldInObject(
+			fieldLabel = "Testing Changed Name",
+			fieldName = "Nickname");
+
+		FormViewBuilder.addFieldToObject(fieldName = "Text");
+
```

Comment:
Hey @nkkwon , thanks for the improvements, I saw that you added a macro to validate if the required option was properly set to the added field in the Form View. However, the most important here is to make sure this property was persisted when reusing this field adding it to a new form view. So, could you please add this validation at the end of the script too? Thanks!